xarray/tests/test_dataarray.py

```diff
@@ -6952,7 +6957,12 @@ def test_drop_duplicates_1d(self, keep) -> None:
         result = da.drop_duplicates("time", keep=keep)
         assert_equal(expected, result)
 
-        with pytest.raises(ValueError, match="['space'] not found"):
+        with pytest.raises(
+            ValueError,
+            match=re.escape(
+                "Dimensions ('space',) not found in dataset dimensions ('time',)"
```

Comment:
This now says "dataset" even though it's a DataArray, because `DataArray.drop_duplicates` just does `self._to_temp_dataset().drop_duplicates(dim, keep=keep)`. Maybe the error message could say "data dimensions"? Should I change it everywhere to say just data instead of dataset?

Reply:
Yes I think not mentioning `Dataset` would be better. There might be an example somewhere else of handling this same type of ambiguity.

Reply:
I recall seeing an error message saying "data dimensions", and I found this: https://github.com/pydata/xarray/blob/main/xarray/core/variable.py#L677 (seems pretty closely related to this PR and #8089). I can change the wording to say just "data dimensions", I think it's a good way to put it.

Reply:
Another option is `"{self.__class__.__name__} dimensions"`, which is used in a couple of places.

Reply:
I changed it now to "data dimensions".